### ISSUE TITLE
recurring stream stalls due to buffers getting dry, vq and aq going down to 0%

### ISSUE THREAD

**Issue by T-C-R** [CastagnaIT/plugin.video.netflix#534]
#### Your Environment
- Netflix add-on version: <plugin.video.netflix v1.1.0 running on Kodi 18.5 x64>
- Operating system version/name: <Ubuntu 18.04.4 LTS, kernel: Linux x86 64-bit version 4.14.24-qnap (QNAP HD-Station)>
- Device model:  <Intel(R) Celeron(R) CPU J1900 @ 1.99GHz, 4 cores available QNAP HS-251+>

Used Operating system:
* [ ] Android
* [ ] iOS
* [x ] Linux
* [ ] OSX
* [ ] Raspberry-Pi
* [ ] Windows

### Describe your help request
<I permanently run into stream stalls, something is drying out the buffer (% fillrate getting down zero, corresponding vq and aq are following). Result is the famous reloading icon indicating the replenshing of the buffer, when 100% is reached streaming continues.
Configuration is with passthrough enabled, but HD 720 resolution selected only. I`m using the option "sync screen to refresh rate", audio delay is compensated by advancedsettings.xml.
Since more than 6months I`m trying to find a systematic, with basically no result. No change of any configuration could significantly change the behavior. Network is not the issue, as Youtube and other streaming portals don`t show the issue at all. Also to run Netflix on the same platform by using Firefox (with 1080p extension even) is no problem.
My suspicion is a combination of decrypting, SW-decoding may cause the a/v running out of sync, or the communication with inputstram.adaptive may cause the problem. How I read the logs always the Audioqueue is running dry first, causing the stream to stall.

I`m attaching a debug-logfile from today for further scrutiny.
[kodi.log](https://github.com/CastagnaIT/plugin.video.netflix/files/4366414/kodi.log)
Frankly, I'm at the end of my pile of ideas and reaching out for further guidance. However, I volunteer to play the testbed in case there are more tests needed.>

#### Screenshots

not yet available, but also not needed as the the issue is explained above.
In case needed for resolution, no problem to add screenshots.

**Comment by CastagnaIT** [comment]
you can't compare netflix to other streaming systems
why they use different servers and different bitrate management

When you have problems with buffer you can check the line with netflix test:
https://fast.com/

i'm sorry but i can't do anything
the flow is not managed by this addon

Usually all the people who have had these problems, after several tests, it is an ISP problem

**Comment by T-C-R** [comment]
Well...
This is a very frustrating answer. I have done thorough testing on my ISP bandwidth, which is a 50Mbit line that, even in busy hours, don´t go below 30Mbit on fast.com. Also, as I wrote in my ticket, with a webbrowser running at 1080p resolution, on the same platform there is no issue. Maybe it is because the inputstream.adaptive cannot deal with real adaptive bitrate management, even at very high bitrates (e.g. movie "our planet"). 
I was hoping to get some more advise where to look at, based on the debug log. Also, there is a nice discussion started at the forum about buffer sizing... ... I will adress the issue to the developer of the inputstream addon then. Thanks.

**Comment by T-C-R** [comment]
think I found the solution... ...by accident. 
It is the audio device configuration. I configured HDMI passthrough on the same device as also for the standard output for multichannel 5.1, and this is confusing Kodi, don´t know why. When I change the latter to another (dead) audiodevice and keep passthrough on the connected HDMI then the aq stays filled at 99% whatever I do. But now I have no sound for AAC, mp3, mp2 and the GUI, as Pulseaudio is no longer working. Once I solved this I´m done.
Found it out myself by studying the debug log thoroughly, lot´s of resyncs of the audio needed before, now not any longer.
